File: programmers/js/level01/17681.js
// [1차] 비밀지도
function solution(n, arr1, arr2) {
  return Array(n).fill(0).map((_, idx) => {
    const decryption = (arr1[idx] | arr2[idx]).toString(2);
    
    return [...'0'.repeat(n - decryption.length) + decryption]
      .reduce((acc, cur) => acc + (+cur ? '#' : ' '), '');
  });
}

// 다른 풀이
// function solution(n, arr1, arr2) {
//   return arr1.map((v, i) => addZero(n, (v | arr2[i]).toString(2)).replace(/1|0/g, a => +a ? '#' : ' '));
// }

// const addZero = (n, s) => {
//   return '0'.repeat(n - s.length) + s;
// }

File: programmers/java/level01/17681.java
// [1차] 비밀지도

import java.util.stream.IntStream;

class Solution {
    String decoding(int num1, int num2, int len) {
        return String.format("%" + len + "s", 
                             Integer.toString(num1 | num2, 2))
                    .replaceAll("1", "#")
                    .replaceAll("0", " ");
    }
    
    public String[] solution(int n, int[] arr1, int[] arr2) {        
        return IntStream.range(0, n)
            .mapToObj(idx -> decoding(arr1[idx], arr2[idx], n))
            .toArray(String[]::new);
    }
}